photon-client/src/main/scala/com/linkedin/photon/ml/avro/AvroIOUtils.scala

```diff
@@ -29,23 +29,42 @@ import org.apache.spark.SparkContext
 import org.apache.spark.rdd.RDD
 
 /**
- * Utility to write Avro files.
+ * Utility to read/write Avro files.
  */
 object AvroIOUtils {
+
   // The upper limit of the file size the for reading single Avro file: 100 MB
   private val READ_SINGLE_AVRO_FILE_SIZE_LIMIT: Long = 100 << 20
 
   /**
+   * Read Avro generic records from the input paths.
+   *
+   * @param sc The Spark context
+   * @param inputPaths The input paths of the generic records
+   * @param minPartitions Minimum number of partitions of the output RDD
+   * @return A [[RDD]] of Avro records of type [[GenericRecord]] read from the specified input paths
+   */
+  protected[ml] def readAvroFiles(
+    sc: SparkContext,
+    inputPaths: Seq[String],
+    minPartitions: Int): RDD[GenericRecord] = {
+
+    assert(inputPaths.nonEmpty, "The number of input paths is zero.")
+    val minPartitionsPerPath = math.ceil(1.0 * minPartitions / inputPaths.length).toInt
+    sc.union(inputPaths.map { path => readOneAvroFile[GenericRecord](sc, path, minPartitionsPerPath) } )
+  }
+
+  /**
    * Read GenericRecord. The output RDD needs to be transformed before any further processing.
    *
    * @param sc Spark context
    * @param inputDir The input directory to the Avro files
    * @param minNumPartitions Minimum number of Hadoop Splits to generate.
    * @return A RDD of records
    */
-  def readFromAvro[T <: GenericRecord : ClassTag](sc: SparkContext, inputDir: String, minNumPartitions: Int): RDD[T] =
+  def readOneAvroFile[T <: GenericRecord : ClassTag](sc: SparkContext, inputDir: String, minNumPartitions: Int): RDD[T] =
```

Comment:
This name is a little misleading, because you could pass a folder path here, in which case multiple physical files would be read.